File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestArrayOptDateValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.ArrayOptDateInValue.ArrayOptDateValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestArrayOptDateValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestArrayOptDateValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                org.joda.time.LocalDate[].class,
                ArrayOptDateValue.class.getDeclaredField("arrayOptDate").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                org.joda.time.LocalDate[].class,
                ArrayOptDateValue.class.getMethod("getArrayOptDate").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                ArrayOptDateValue.class,
                ArrayOptDateValue.class.getMethod("setArrayOptDate", org.joda.time.LocalDate[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertArrayEquals(
                new org.joda.time.LocalDate[0],
                new ArrayOptDateValue().getArrayOptDate());
    }
}

File: legacy/workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestArrayOptDateValue.scala
package com.dslplatform.ocd.values

import ArrayOptDateInValue.ArrayOptDateValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestArrayOptDateValue extends SpecExtensions {
  def `Test property field type: Array[Option[org.joda.time.LocalDate]]`() =
    checkType(
      typeOf[Array[Option[org.joda.time.LocalDate]]]
    , typeOf[ArrayOptDateValue].member("arrayOptDate": TermName).asMethod.returnType
    )

  def `Test property default value: Array.empty[Option[org.joda.time.LocalDate]]`() {
    assert(ArrayOptDateValue().arrayOptDate === Array.empty[Option[org.joda.time.LocalDate]])
  }
}
